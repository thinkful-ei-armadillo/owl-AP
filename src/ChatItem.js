import React from 'react';
import './ChatItem.css';

export default function ChatItem(props) {
  let itemHTML;
  let type = props.type;
  if (type === 'message') {
    itemHTML = (
      <li className="session-participant">
        <img src={props.user.avatar} alt="user img" />
        <div className="participant-info">
          <h6>{props.user.name}</h6>
          <div className="chat-message">{props.message}</div>
        </div>
      </li>
    );
  } else {
    let msg = `${props.user.name}`;
    if (type === 'thumbs-up') {
      msg += ' gave a thumbs up';
    } else if (type === 'thumbs-down') {
      msg += ' gave a thumbs down';
    } else if (type === 'raise-hand') {
      msg += ' raises their hand';
    } else if (type === 'clap') {
      msg += ' claps';
    } else if (type === 'join') {
      msg += ' joins';
    } else if (type === 'join-stage') {
      msg += ' joins the stage';
    } else if (type === 'leave') {
      msg += ' leave';
    } else if (type === 'leave-stage') {
      msg += ' leaves the stage';
    }

    itemHTML = (
      <li className="session-participant">
        <div className="participant-info">
          <div className="chat-message">{msg}</div>
        </div>
      </li>
    );
  }

  return itemHTML;
}
