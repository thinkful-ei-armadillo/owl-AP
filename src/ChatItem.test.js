import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatItem from './ChatItem';
import store from './store';

describe('Participant component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(
      <ChatItem
      key={1}
      user={{name: 'Ronnie', avatar: 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'}}
      type='message'
      message='hello world'
      time={1548852646559}
      timestamp={1548852646559}
      />,
      div
    );

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <ChatItem
      key={1}
      user={{name: 'Ronnie', avatar: 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'}}
      type='message'
      message='hello world'
      time={1548852646559}
      timestamp={1548852646559}
      />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
