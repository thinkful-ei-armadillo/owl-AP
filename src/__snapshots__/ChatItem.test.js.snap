// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant component testing renders the UI as expected 1`] = `
<li
  className="session-participant"
>
  <img
    alt="user img"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <div
    className="participant-info"
  >
    <h6>
      Ronnie
    </h6>
    <div
      className="chat-message"
    >
      hello world
    </div>
  </div>
</li>
`;
